core/domain/collection_domain_test.py

```diff
@@ -460,6 +460,52 @@ def test_next_explorations(self):
             collection.get_next_exploration_ids(
                 ['exp_id_0', 'exp_id_1', 'exp_id_2', 'exp_id_3']), [])
 
+    def test_next_explorations_in_sequence(self):
+
+        collection = collection_domain.Collection.create_default_collection(
+            'collection_id')
+        exploration_id = 'exp_id_0'
+        collection.add_node(exploration_id)
+
+        # Completing the only exploration of the collection should lead to no
+        # available explorations thereafter.
+        self.assertEqual(
+            collection.get_next_exploration_ids_in_sequence(exploration_id), [])
+
+        # If the current exploration has no acquired skills, a list of general
+        # explorations should be returned.
+        collection.add_node('exp_id_1')
+        collection.add_node('exp_id_2')
+        self.assertEqual(
+            collection.get_next_exploration_ids_in_sequence(exploration_id),
+            ['exp_id_1', 'exp_id_2'])
+
+        # If only one exploration in the collection has a prerequisite skill
+        # that is included in the user's learned skills, only that exploration
+        # should be returned.
+        collection_node0 = collection.get_node('exp_id_0')
+        collection_node2 = collection.get_node('exp_id_2')
+
+        collection_node0.update_acquired_skills(['skill1a'])
+        collection_node2.update_prerequisite_skills(['skill1a'])
+        self.assertEqual(
+            collection.get_next_exploration_ids_in_sequence(exploration_id),
+            ['exp_id_2'])
+
+        # An exploration should not be suggested unless all of its prerequisite
+        # skills have been learned.
+        collection_node2.update_prerequisite_skills(['skill1a', 'skill1b'])
+        self.assertNotIn(
+            'ex_id_2',
+            collection.get_next_exploration_ids_in_sequence(exploration_id))
+
+        # If the current node's prerequisite skills include one of it's
+        # acquired skills, the current node should not be returned in the list.
+        collection_node0.update_prerequisite_skills(['skill1a'])
+        self.assertNotIn(
+            exploration_id,
+            collection.get_next_exploration_ids_in_sequence(exploration_id))
```

Comment:
Added test. The controller logic still needs to be added but I'm having trouble getting a list of general explorations to show up at all on my localhost.  For example, I've loaded all the explorations in the admin panel but regardless of my logged in status, when I access 'The Solar System' exploration outside of its collection there are no recommendations returned when I complete it.  Shouldn't a general list of explorations show up in this case?